diff --git a/Mac/BuildScript/build-installer.py b/Mac/BuildScript/build-installer.py
index b31cb76..3d560bf 100755
--- a/Mac/BuildScript/build-installer.py
+++ b/Mac/BuildScript/build-installer.py
@@ -314,9 +314,9 @@ def library_recipes():
     if PYTHON_3:
         result.extend([
           dict(
-              name="XZ 5.2.3",
-              url="http://tukaani.org/xz/xz-5.2.3.tar.gz",
-              checksum='ef68674fb47a8b8e741b34e429d86e9d',
+              name="XZ 5.2.10",
+              url="https://github.com/tukaani-project/xz/releases/download/v5.2.10/xz-5.2.10.tar.gz",
+              checksum='eae070cfcfec883a91866b4fe37cce17',
               configure_pre=[
                     '--disable-dependency-tracking',
               ]
@@ -326,7 +326,7 @@ def library_recipes():
     result.extend([
           dict(
               name="NCurses 6.5",
-              url="https://ftp.gnu.org/gnu/ncurses/ncurses-6.5.tar.gz",
+              url="https://invisible-island.net/archives/ncurses/ncurses-6.5.tar.gz",
               checksum="136d91bc269a9a5785e5f9e980bc76ab57428f604ce3e5a5a90cebc767971cc6",
               configure_pre=[
                   "--datadir=/usr/share",
